shared/src/native-src/util.cpp

```diff
@@ -209,7 +209,7 @@ namespace shared
 #ifdef _WIN32
         return SetEnvironmentVariable(::shared::Trim(name).c_str(), value.c_str());
 #else
-        return setenv(::shared::ToString(name).c_str(), ::shared::ToString(value).c_str(), 1) == 1;
+        return setenv(::shared::ToString(name).c_str(), ::shared::ToString(value).c_str(), 1) == 0;
```

Comment:
I take it all the changes in util.cpp and string.cpp are bugs you found from the testing! Hooray! and Eek! 😅

Reply:
exactly :)

Reply:
For this specific one, the `man` says that 0 is a success value https://man7.org/linux/man-pages/man3/setenv.3.html